forge-gui/res/cardsfolder/upcoming/traumatic_prank.txt

```diff
@@ -0,0 +1,9 @@
+Name:Traumatic Prank
+ManaCost:2 R
+Types:Sorcery
+A:SP$ GainControl | ValidTgts$ Creature | TgtPrompt$ Select target creature | LoseControl$ EOT | AddKWs$ Haste | Untap$ True | SubAbility$ DBEffect | SpellDescription$ Gain control of target creature until end of turn. Untap that creature. It perpetually gains haste, "This creature can't block," and "At the beginning of your upkeep, this creature deals 1 damage to you."
+SVar:DBEffect:DB$ Effect | RememberObjects$ Targeted | StaticAbilities$ TraumaticPrank | Name$ Traumatic Prank's Perpetual Effect | Duration$ Permanent
+SVar:TraumaticPrank:Mode$ Continuous | Affected$ Card.IsRemembered | AddHiddenKeyword$ CARDNAME can't block. | AddKeyword$ Haste | AddTrigger$ PingUpkeep | EffectZone$ Command | AffectedZone$ Battlefield,Hand,Graveyard,Exile,Stack,Library,Command | Description$ Creature perpetually gains haste, "This creature can't block," and "At the beginning of your upkeep, this creature deals 1 damage to you."
+SVar:PingUpkeep:Mode$ Phase | Phase$ Upkeep | ValidPlayer$ You | TriggerZones$ Battlefield | Execute$ TrigDamage | TriggerDescription$ At the beginning of your upkeep, CARDNAME deals 1 damage to that player.
```

Comment:
make `TriggerDescription$ At the beginning of your upkeep, this creature deals 1 damage to you.`

This ability is gross because it is perpetual, but I'm pretty sure your current script is correct in that the creature is not in fact dealing damage from the graveyard, exile, etc. Another case where it would be nice if they actually thought through or better explained the rules implications.